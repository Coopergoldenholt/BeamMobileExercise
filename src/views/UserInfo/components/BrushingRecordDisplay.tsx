import { duration } from 'moment'
import React from 'react'
import { TouchableOpacity, Text, View, StyleSheet } from 'react-native'
import { formatDateTime } from '../../../functions/formatDateTime'
import { IBrushingRecord } from '../../../types'

interface BrushingRecordProps {
    record: IBrushingRecord;
    navigation: any;
    alertBrushingRecord: any;
}

const BrushingRecordDisplay: React.FC<BrushingRecordProps> = ({ record, navigation, alertBrushingRecord }) => {
    const { date, duration, id } = record

    return (
        <TouchableOpacity
            testID='Navigate.Button'
            style={styles.brushingRecord}
            onPress={() => navigation.navigate('BrushingRecord', { record })}>
            <Text style={styles.dateText}>{formatDateTime(date)}</Text>
            <Text>Duration:</Text>
            <View style={styles.durationContainer}>
                <Text>Minutes: {duration.minutes}</Text>
                <Text>Seconds: {duration.seconds}</Text>
            </View>
            <TouchableOpacity style={styles.deleteButton} onPress={() => alertBrushingRecord(id)}>
                <Text>Delete</Text>
            </TouchableOpacity>
        </TouchableOpacity>
    )
}


const styles = StyleSheet.create({
    container: {
        padding: 10
    },
    brushingRecord: {
        paddingTop: 15,
        borderColor: 'grey',
        borderTopWidth: 1
    },
    button: {
        alignSelf: 'center',
        backgroundColor: '#20a9ff',
        borderRadius: 5,
        height: 40,
        width: 200,
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: 30,
        marginBottom: 20
    },
    buttonText: {
        color: 'white'
    },
    durationContainer: {
        flexDirection: 'row',
        justifyContent: 'space-around'
    },
    dateText: {
        fontSize: 20,
        paddingBottom: 10
    },

    deleteButton: {
        alignSelf: 'center',
        backgroundColor: 'tomato',
        borderRadius: 5,
        height: 20,
        width: 100,
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: 10,
        marginBottom: 30
    }
})

export default BrushingRecordDisplay