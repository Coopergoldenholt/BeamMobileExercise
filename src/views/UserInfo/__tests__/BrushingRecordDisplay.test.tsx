import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import BrushingRecordDisplay from '../components/BrushingRecordDisplay'

let record = {
    date: new Date(2018, 11, 24, 10, 33, 30, 0),
    duration: {
        minutes: '2',
        seconds: '32'
    },
    id: 'as8df9hawionr2w'
}

let navigation = {
    navigate: jest.fn()
}
const alertBrushingRecord = jest.fn()


test('renders correctly', () => {

    const { toJSON } = render(<BrushingRecordDisplay
        alertBrushingRecord={alertBrushingRecord}
        record={record}
        navigation
    />)
    expect(toJSON()).toMatchSnapshot();
})

test('it navigates on press', () => {
    const { getByTestId } = render(<BrushingRecordDisplay
        alertBrushingRecord={alertBrushingRecord}
        record={record}
        navigation={navigation}
    />)

    fireEvent.press(getByTestId('Navigate.Button'))
    expect(navigation.navigate).toBeCalledWith('BrushingRecord', { record })
})

test('it calls delete button', () => {
    const { getByText } = render(<BrushingRecordDisplay
        alertBrushingRecord={alertBrushingRecord}
        record={record}
        navigation={navigation}
    />)
    fireEvent.press(getByText('Delete'))
    expect(alertBrushingRecord).toBeCalled()
})

test('it renders correct time', () => {
    const { getByText } = render(<BrushingRecordDisplay
        alertBrushingRecord={alertBrushingRecord}
        record={record}
        navigation={navigation}
    />)

    expect(getByText('Minutes: 2')).not.toBeNull()
    expect(getByText('Seconds: 32')).not.toBeNull()
})