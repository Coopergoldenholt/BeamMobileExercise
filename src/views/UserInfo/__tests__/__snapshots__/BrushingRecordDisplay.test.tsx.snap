// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "grey",
      "borderTopWidth": 1,
      "opacity": 1,
      "paddingTop": 15,
    }
  }
  testID="Navigate.Button"
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "paddingBottom": 10,
      }
    }
  >
    December 24, 2018 10:33 AM
  </Text>
  <Text>
    Duration:
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <Text>
      Minutes: 
      2
    </Text>
    <Text>
      Seconds: 
      32
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "tomato",
        "borderRadius": 5,
        "height": 20,
        "justifyContent": "center",
        "marginBottom": 30,
        "marginTop": 10,
        "opacity": 1,
        "width": 100,
      }
    }
  >
    <Text>
      Delete
    </Text>
  </View>
</View>
`;
